src/main/java/seedu/bankwithus/BankWithUs.java

```diff
@@ -40,18 +46,34 @@ public BankWithUs(String filePath) throws IOException {
      *
      * @throws IOException throw error if the data cannot be saved
      */
-    private void exit(String filePath) throws IOException {
+    public void exit(String filePath) throws IOException {
         try {
             storage.saveToFile(accounts);
         } catch (IOException e) {
             ui.showIOError();
             throw e;
         }
         ui.showFarewellMessage();
+        ui.closeScanner();
     }
 
-    public void run() {
-
+    /**
+     * The main command and output loop. Takes in user input line by line
+     * and gives it to the parser to execute the command.
+     * 
+     * @throws IOException if something goes wrong while exiting the program
+     */
+    public void run() throws IOException {
+        ui.createScanner();
+        while (!isExitEntered) {
+            String line = ui.getNextLine();
+            try {
+                parser.parseUserInput(line);
+            } catch (CommandNotFoundException e) {
+                ui.showCommandNotFoundError();
+            }
+        }
```

Comment:
Oh shit, I just realised. I should have let parseUserInput() call exit(), and then since exit() is a function in BWU, exit can modify the value of isExitEntered directly, which means that isExitEntered wont need to be a public function. Will do once vishnu is done with his current pull request